newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -26,53 +27,51 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
 				Type:        schema.TypeString,
 				Required:    true,
 				ForceNew:    true,
-				Description: "The monitor type. Valid values are SIMPLE, BROWSER, SCRIPT_BROWSER, and SCRIPT_API.",
+				Description: "The monitor type. Valid values are SIMPLE AND BROWSER.",
 				ValidateFunc: validation.StringInSlice([]string{
 					"SIMPLE",
 					"BROWSER",
-					"SCRIPT_API",
-					"SCRIPT_BROWSER",
 				}, false),
 			},
 			"name": {
 				Type:        schema.TypeString,
 				Required:    true,
 				Description: "The title of this monitor.",
 			},
-			"frequency": {
-				Type:         schema.TypeInt,
-				Required:     true,
-				ValidateFunc: intInSlice([]int{1, 5, 10, 15, 30, 60, 360, 720, 1440}),
-				Description:  "The interval (in minutes) at which this monitor should run. Valid values are 1, 5, 10, 15, 30, 60, 360, 720, or 1440.",
+			"period": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "The interval at which this monitor should run. Valid values are EVERY_MINUTE, EVERY_5_MINUTES, EVERY_10_MINUTES, EVERY_15_MINUTES, EVERY_30_MINUTES, EVERY_HOUR, EVERY_6_HOURS, EVERY_12_HOURS, or EVERY_DAY.",
+				ValidateFunc: validation.StringInSlice(listValidSyntheticsMonitorPeriods(), false),
 			},
 			"uri": {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "The URI for the monitor to hit.",
 				// TODO: ValidateFunc (required if SIMPLE or BROWSER)
 			},
-			"locations": {
+
+			// TODO: Locations needs to include both private and public
+			"locations_public": {
 				Type:        schema.TypeSet,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				MinItems:    1,
 				Required:    true,
```

Comment:
`locations_public` field is required, and `locations_private` is not. This means I cannot create a synthetics check that only runs on private locations as I will always need to set a `locations_public`. Can we check if that's correct? I would assume you can run a synthetics check on only private locations.

Reply:
Could use `AtLeastOneOf` so that one is set [https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.17.0/helper/schema/schema.go#L298](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.17.0/helper/schema/schema.go#L298)